classes/socket.h

```diff
@@ -236,16 +236,17 @@ PHP_METHOD(Socket, accept) {
 	pthreads_socket_accept(getThis(), ce, return_value);
 } /* }}} */
 
-/* {{{ proto bool Socket::connect(string host, int port) */
+/* {{{ proto bool Socket::connect(string host[, int port]) */
 PHP_METHOD(Socket, connect) {
 	zend_string *host = NULL;
 	zend_long port = 0;
+	int argc = ZEND_NUM_ARGS();
 
-	if (zend_parse_parameters(ZEND_NUM_ARGS(), "Sl", &host, &port) != SUCCESS) {
-		return;
+	if (zend_parse_parameters(argc, "S|l", &host, &port) != SUCCESS) {
+		RETURN_FALSE;
```

Comment:
This seems redundant, isn't it usual to just `return;` here? never see anything else do this

Reply:
Socket::connect defines an explicit return type, so a bool value is here required too.